### ISSUE TITLE
Running nx migrate for an 12.10.1 workspace to 15.1.1 fails (angular)

### ISSUE THREAD

**Issue by bameyrick** [nrwl/nx#13976]
### Current Behavior

Running nx migrate for an 12.10.1 workspace (angular 12.2.0) to 15.1.1 (for angular ~14.2.0) fails with the error

```
ERROR  The migrate command failed.

Package subpath './package.json' is not defined by "exports" in C:\Users\<USERNAME>\AppData\Local\Temp\tmp-12908-9ftAt8X4v93C\node_modules\postcss-preset-env\package.json
```

This is using node 18.12.1 on windows but I have also tried using 16.10.0 and 14.15.0.

### Expected Behavior

The migrate command should update the project's package.json and create a migrations.json file.

### Github Repo

https://github.com/bameyrick/nx-migrate-postcss-preset-env-error

### Steps to Reproduce

1. Clone repository
2. Install dependencies
3. Install `nx@12.10.1` globally
4. Run `nx migrate 15.1.1` using cmd.exe running as administrator
5. Eventually migration should fail and error will be shown

### Nx Report

```shell
Node : 18.12.1
 OS   : win32 x64
 npm  : 8.19.2

 nx : 12.10.1
 @nrwl/angular : 12.10.1
 @nrwl/cli : 12.10.1
 @nrwl/cypress : 12.10.1
 @nrwl/devkit : 12.10.1
 @nrwl/eslint-plugin-nx : 12.10.1
 @nrwl/express : Not Found
 @nrwl/jest : 12.10.1
 @nrwl/linter : 12.10.1
 @nrwl/nest : Not Found
 @nrwl/next : Not Found
 @nrwl/node : Not Found
 @nrwl/nx-cloud : Not Found
 @nrwl/react : Not Found
 @nrwl/schematics : Not Found
 @nrwl/tao : 12.10.1
 @nrwl/web : Not Found
 @nrwl/workspace : 12.10.1
 @nrwl/storybook : 12.10.1
 @nrwl/gatsby : Not Found
 typescript : 4.3.5
```

### Failure Logs

```shell
ERROR  The migrate command failed.

Package subpath './package.json' is not defined by "exports" in C:\Users\<username>\AppData\Local\Temp\tmp-12908-9ftAt8X4v93C\node_modules\postcss-preset-env\package.json
C:\Users\<username>\projects\migrate-postcss-preset-env-error\node_modules\yargs\build\lib\yargs.js:1132
                throw err;
                ^

Error: Command failed: npx tao migrate 15.1.1
    at checkExecSyncError (node:child_process:871:11)
    at Object.execSync (node:child_process:943:15)
    at Object.handler (C:\Users\<username>\projects\migrate-postcss-preset-env-error\node_modules\@nrwl\workspace\src\command-line\nx-commands.js:107:25)
    at Object.runCommand (C:\Users\<username>\projects\migrate-postcss-preset-env-error\node_modules\yargs\build\lib\command.js:196:48)
    at Object.parseArgs [as _parseArgs] (C:\Users\<username>\projects\migrate-postcss-preset-env-error\node_modules\yargs\build\lib\yargs.js:1043:55)
    at Object.get [as argv] (C:\Users\<username>\projects\migrate-postcss-preset-env-error\node_modules\yargs\build\lib\yargs.js:986:25)
    at Object.initLocal (C:\Users\<username>\projects\migrate-postcss-preset-env-error\node_modules\@nrwl\cli\lib\init-local.js:25:79)
    at Object.<anonymous> (C:\Users\<username>\projects\migrate-postcss-preset-env-error\node_modules\@nrwl\cli\bin\nx.js:43:18)
    at Module._compile (node:internal/modules/cjs/loader:1159:14) {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 20736,
  stdout: null,
  stderr: null
}
```

**Comment by AgentEnder** [comment]
Hey @bameyrick! We only support migrating 1 major version at a time? Can you try to migrate to the latest v13 and v14 releases first? 

Thanks.

**Comment by bameyrick** [comment]
Hi @AgentEnder, I've tried migrating to `13.1.4` and then `14.1.9` (was looking at the recommended nx versions here https://nx.dev/packages/angular/documents/angular-nx-version-matrix) and got the same error during `nx migrate 14.1.9`.

I then tried to migrate to latest 13 (`13.10.6`) and got the error `Cannot find module '../adapter/ngcli-adapter'` when running `nx migrate --run-migrations`.

Are there any specific versions I should try migrating to?
